tests/mesh/extra_integers.C

```diff
@@ -0,0 +1,100 @@
+// Ignore unused parameter warnings coming from cppunit headers
+#include <libmesh/ignore_warnings.h>
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/TestCase.h>
+#include <libmesh/restore_warnings.h>
+
+#include <libmesh/libmesh.h>
+#include <libmesh/mesh.h>
+#include <libmesh/elem.h>
+#include <libmesh/mesh_generation.h>
+
+#include "test_comm.h"
+
+// THE CPPUNIT_TEST_SUITE_END macro expands to code that involves
+// std::auto_ptr, which in turn produces -Wdeprecated-declarations
+// warnings.  These can be ignored in GCC as long as we wrap the
+// offending code in appropriate pragmas.  We can't get away with a
+// single ignore_warnings.h inclusion at the beginning of this file,
+// since the libmesh headers pull in a restore_warnings.h at some
+// point.  We also don't bother restoring warnings at the end of this
+// file since it's not a header.
+#include <libmesh/ignore_warnings.h>
+
+using namespace libMesh;
+
+class ExtraIntegersTest : public CppUnit::TestCase
+{
+  /**
+   * The goal of this test is to verify the ability to add extra
+   * integer storage to a Mesh object, then set and query those extra
+   * integers in the objects within the mesh.
+   */
+public:
+  CPPUNIT_TEST_SUITE( ExtraIntegersTest );
+
+  CPPUNIT_TEST( testExtraIntegersEdge2 );
+
+  CPPUNIT_TEST_SUITE_END();
+
+protected:
+  // Helper function called by the test implementations, saves a few lines of code.
+  void test_helper_1D(ElemType elem_type)
+  {
+    Mesh mesh(*TestCommWorld, /*dim=*/2);
+
+    // Request some extra integers before building
+    unsigned int i1 = mesh.add_extra_integer("i1");
```

Comment:
Should this be `add_elem_integer`?

Reply:
Not sure if a function like `get_extra_integer_index(const std::string &)` will be useful. I guess it does not hurt to add it.

Reply:
Yeah, this is one of the completely broken things I'm up late fixing right now.  :-)

Reply:
get_elem_integer_index would be helpful for debugging?  add_extra_integer *already* gives that functionality, since you can call it multiple times with the same name and it'll return the same index value, but doing so is dangerous, since following ```add_extra_integer("correct_name")``` with ```add_extra_integer("corect_name")``` will just silently add a new integer with the wrong name and return the index for that.  With a "get_elem_integer_index" we could actually assert that the requested name exists.

Reply:
`get_elem_integer_index` can be const. It was in my mind that adding the integer is in the mesh generation, while the integer can be used in a material. The material does not needs to know the integer, but hard-coding a name for the integer is better. Actually this may not be important because I can find a place to store the index and have a meaningful function such as `getMaterialIDIndex` to retrieve it.